app/src/main/java/com/example/myapplication/network/ApiBuilder.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.myapplication.network
+
+import com.example.myapplication.BuildConfig
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.http.GET
+
+class ApiBuilder(private val baseUrl: String) {
+
+    private val okHttpClient: OkHttpClient
+        get() {
+            val loggingInterceptor = HttpLoggingInterceptor().apply {
+                level =
+                    if (BuildConfig.DEBUG) HttpLoggingInterceptor.Level.BODY else HttpLoggingInterceptor.Level.BASIC
+            }
+            return OkHttpClient.Builder()
+                .addInterceptor(loggingInterceptor)
+                .build()
+        }
+
+    fun <T> create(serviceInterface: Class<T>): T {
+        val retrofit = Retrofit.Builder()
+            .baseUrl(baseUrl)
```

Comment:
This URL will (or should) never change. Having that in mind, you can add a BuildConfig field with that base URL, just like it's done [here](https://developer.android.com/studio/build/gradle-tips#share-custom-fields-and-resource-values-with-your-app-code).